config/src/network_id.rs

```diff
@@ -81,12 +81,69 @@ impl NetworkContext {
 /// and handshakes should verify that the NetworkId being used is the same during a handshake,
 /// to effectively ensure communication is restricted to a network.  Network should be checked that
 /// it is not the `DEFAULT_NETWORK`
-#[derive(Clone, Deserialize, Eq, Hash, PartialEq, Serialize)]
-#[serde(rename = "NetworkId", rename_all = "snake_case")]
+#[derive(Clone, Copy, Eq, Hash, PartialEq)]
 pub enum NetworkId {
     Validator,
     Public,
-    Private(String),
+    /// An invalid type due to migration from old NetworkId type to new type.  This will *not* deserialize
+    /// TODO: Once fully migrated, this could become Vfn through migration
+    Invalid,
```

Comment:
@bmwill do you know if there's a way to have like an empty slot in the enum, but we need the new Vfn variant to not use the old slot? I guess if this is a dumb enum now, you might be able to do something like

```rust
enum NetworkId {
    Validator = 0,
    Public = 1,
    Vfn = 3,
}
```

(can't remember exactly how that works)

Reply:
I tried that, it didn't work :(

Reply:
😭

oh well, it's probably fine like this for now

Reply:
you need something like #[repr(u8)]
```
#[repr(u8)]
enum Foo {
    A = 1,
    B = 3,
    C = 5,
    D = 7,
}
```

Reply:
I swear I tried both of these earlier and neither worked.  Now they both work...